lib/default.nix

```diff
@@ -148,6 +148,52 @@ let
           _module.args.perSystem = systemArgs.${pkgs.system}.perSystem;
         };
 
+      homes =
+        let
+          loadHomeConfiguration =
+            path: pkgs:
+            inputs.home-manager.lib.homeManagerConfiguration {
+              inherit pkgs;
+              extraSpecialArgs = specialArgs;
+              modules = [
+                perSystemModule
+                path
+              ];
+            };
+
+          loadDefault = path: pkgs: (import path) { inherit pkgs inputs flake; };
+
+          mkUserConfig =
+            hostname: username: entry: pkgs:
+            if entry.type == "regular" then
+              loadHomeConfiguration entry.path pkgs
+            else if builtins.pathExists (entry.path + "/default.nix") then
+              loadDefault (entry.path + "/default.nix") pkgs
+            else if builtins.pathExists (entry.path + "/home-configuration.nix") then
+              loadHomeConfiguration (entry.path + "/home-configuration.nix") pkgs
+            else
+              throw "home '${username}@${hostname}' does not have a configuration";
+
+          mkHostUsers =
+            hostname: entries:
+            let
+              validUsers = lib.filterAttrs (_: { type, ... }: type == "regular" || type == "directory") entries;
+            in
+            lib.concatMapAttrs (username: entry: {
+              "${username}@${hostname}" = mkUserConfig hostname username entry;
+            }) validUsers;
+
+          mkHosts =
+            entries:
+            let
+              hostDirs = lib.filterAttrs (_: { type, ... }: type == "directory") entries;
+            in
+            lib.concatMapAttrs (
+              hostname: { path, ... }: importDir (path + "/home") (mkHostUsers hostname)
```

Comment:
I like the file layout overall, it follows the same patterns as the rest.

It might be worth renaming the folder from "home" to "users" to keep the generic hosts / users mapping.